src/types/src/obs.ts

```diff
@@ -0,0 +1,192 @@
+	SET_PREVIEW_SCENE = 'SetPreviewScene',
+	SET_CURRENT_TRANSITION = 'SetCurrentTransition',
+	START_RECORDING = 'StartRecording',
+	STOP_RECORDING = 'StopRecording',
+	START_STREAMING = 'StartStreaming',
+	STOP_STREAMING = 'StopStreaming',
+	SET_SCENE_ITEM_RENDEER = 'SetSceneItemRender',
+	SET_MUTE = 'SetMute',
+	SET_SOURCE_SETTINGS = 'SetSourceSettings'
+}
+
+export type TimelineObjOBSAny =
+	| TimelineObjOBSCurrentScene
+	| TimelineObjOBSCurrentTransition
+	| TimelineObjOBSRecording
+	| TimelineObjOBSStreaming
+	| TimelineObjOBSSceneItemRender
+	| TimelineObjOBSMute
+	| TimelineObjOBSSourceSettings
+
+export enum TimelineContentTypeOBS {
+	CURRENT_SCENE = 'CURRENT_SCENE',
+	CURRENT_TRANSITION = 'CURRENT_TRANSITION',
+	RECORDING = 'RECORDING',
+	STREAMING = 'STREAMING',
+	SCENE_ITEM_RENDER = 'SCENE_ITEM_RENDER',
+	MUTE = 'MUTE',
+	SOURCE_SETTINGS = 'SOURCE_SETTINGS'
+}
+export interface TimelineObjOBSBase extends TSRTimelineObjBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS;
+	}
+}
+
+export interface TimelineObjOBSCurrentScene extends TimelineObjOBSBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS.CURRENT_SCENE;
+
+		/** Name of the scene that should be current */
+		sceneName: string;
+	}
+}
+
+export interface TimelineObjOBSCurrentTransition extends TimelineObjOBSBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS.CURRENT_TRANSITION;
+
+		/** Name of the transition that should be current */
+		transitionName: string;
+	}
+}
+
+export interface TimelineObjOBSRecording extends TimelineObjOBSBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS.RECORDING;
+
+		/** Should recording be turned on */
+		on: boolean;
+	}
+}
+
+export interface TimelineObjOBSStreaming extends TimelineObjOBSBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS.STREAMING;
+
+		/** Should streaming be turned on */
+		on: boolean;
+	}
+}
+
+export interface TimelineObjOBSSceneItemRender extends TimelineObjOBSBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS.SCENE_ITEM_RENDER;
+
+		/** Should the scene item be enabled */
+		on: boolean;
+	}
+}
+
+type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never }
+type XOR<T, U> = (T | U) extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U
+
+export interface TimelineObjOBSSourceSettings extends TimelineObjOBSBase {
+	content: {
+		deviceType: DeviceType.OBS;
+		type: TimelineContentTypeOBS.SOURCE_SETTINGS;
+	} & XOR<{
+		sourceType: 'ffmpeg_source',
+		sourceSettings: {
+			close_when_inactive?: boolean,
+			hw_decode?: boolean
+			input?: string
+			is_local_file?: boolean
+			local_file?: string
+			looping?: boolean
+		}
+	}, {
+		sourceType: 'dshow_input' | 'browser_source' | 'window_capture' | 'image_source'
+	}>
+}
+
+export interface TimelineObjOBSMute extends TimelineObjOBSBase {
```

Comment:
I am wondering if this should be `TimelineObjOBSSourceAudio` instead of being just for muted, as there are other audio properties that I assume can be set over websocket, and it might be nice to be able to group inside of single timeline objects

Reply:
Yeah, good point. I just took commands and converted them into timeline objects, but I agree that it makes sense to add volume here as well.

Reply:
Actually, after a bit of a think, I think I'm gonna keep it that way. The other properties that are available are volume faders and channel assignments. While I guess it may be useful to automate channel assignments, I suspect these changes will be separate from Mute/Unmute and Volume operations. I also think volume manipulation should also probably (eventually) live in it's own timeline object. My reasoning behind this is that with mute, you can almost use a semi-Sisyphos workflow, just putting audio sources on/off air, while having full manual control over their volume. Having volume automation alongside muting will invite using volume for audio automation and effectively invite the sort of human-machine conflicts Sofie allows to avoid. So I'll be keeping mute in a separate object, as that seems to be to be more practical.